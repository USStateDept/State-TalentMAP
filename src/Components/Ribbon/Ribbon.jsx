import PropTypes from 'prop-types';
import FA from 'react-fontawesome';

const Ribbon = ({ type, className, icon, text, cutSide, containerProps, isWide, showText,
  isWideResults }) => (
  <div
    className={`ribbon-outer-container-cut-${cutSide} ${isWide ? 'ribbon-wide' : ''} ${isWideResults ? 'ribbon-wide-results' : ''} ${className}`}
    {...containerProps}
  >
    <div className={`ribbon ribbon-${type} ribbon-cut-${cutSide}`}>
      <FA name={icon} />
      {
        showText &&
      <span className="text">{text}</span>
      }
    </div>
  </div>
);


Ribbon.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.string,
  text: PropTypes.string,
  containerProps: PropTypes.shape({}),
  type: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'success', 'hds', 'snd', 'htf']),
  cutSide: PropTypes.oneOf(['left', 'right', 'both']),
  isWide: PropTypes.bool,
  showText: PropTypes.bool,
  isWideResults: PropTypes.bool,
};

Ribbon.defaultProps = {
  className: '',
  icon: 'handshake',
  text: '',
  containerProps: {},
  type: 'primary',
  cutSide: 'left',
  isWide: false,
  showText: true,
  isWideResults: false,
};

export default Ribbon;
