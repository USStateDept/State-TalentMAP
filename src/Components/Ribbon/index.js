import Featured from './Featured';
import Handshake from './Handshake';
import HistDiffToStaff from './HistDiffToStaff';
import ServiceNeedDifferential from './ServiceNeedDifferential';
import InBidList from './InBidList';
import IsHardToFill from './IsHardToFill';

export { Handshake, HistDiffToStaff,
  ServiceNeedDifferential, Featured, InBidList, IsHardToFill };

export { default } from './Ribbon';
