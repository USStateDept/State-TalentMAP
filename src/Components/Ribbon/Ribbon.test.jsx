import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Ribbon from './Ribbon';

describe('RibbonComponent', () => {
  const props = {
    type: 'primary',
    className: '',
    icon: '',
    text: '',
    cutSide: 'left',
    containerProps: {},
  };

  it('is defined', () => {
    const wrapper = shallow(<Ribbon {...props} />);

    expect(wrapper).toBeDefined();
  });

  it('creates class names', () => {
    const props$ = {
      ...props,
      className: 'custom-class',
      cutSide: 'right',
      type: 'primary',
    };
    const wrapper = shallow(<Ribbon {...props$} />);

    expect(wrapper.find('div').at(0).props().className)
      .toBe(`ribbon-outer-container-cut-${props$.cutSide}   ${props$.className}`);
    expect(wrapper.find('div').at(1).props().className)
      .toBe(`ribbon ribbon-${props$.type} ribbon-cut-${props$.cutSide}`);
  });

  it('passes the correct icon name', () => {
    const props$ = {
      ...props,
      icon: 'message',
    };
    const wrapper = shallow(<Ribbon {...props$} />);

    expect(wrapper.find('FontAwesome').at(0).props().name)
      .toBe(props$.icon);
  });

  it('passes the correct text', () => {
    const props$ = {
      ...props,
      text: 'Ribbon text',
    };
    const wrapper = shallow(<Ribbon {...props$} />);

    expect(wrapper.find('span').text())
      .toBe(props$.text);
  });

  it('spreads the containerProps', () => {
    const props$ = {
      ...props,
      containerProps: {
        alt: 'alt text',
        tabIndex: 0,
      },
    };
    const wrapper = shallow(<Ribbon {...props$} />);

    Object.keys(props$.containerProps).map(m => (
      expect(wrapper.find('div').at(0).props()[m])
        .toBe(props$.containerProps[m])
    ));
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Ribbon {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
