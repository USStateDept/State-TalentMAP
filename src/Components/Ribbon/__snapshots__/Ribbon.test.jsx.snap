// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RibbonComponent matches snapshot 1`] = `
<div
  className="ribbon-outer-container-cut-left   "
>
  <div
    className="ribbon ribbon-primary ribbon-cut-left"
  >
    <FontAwesome
      name=""
    />
    <span
      className="text"
    />
  </div>
</div>
`;
